src/library/scala/util/Random.scala

```diff
@@ -65,11 +88,27 @@ class Random(val self: java.util.Random) extends AnyRef with Serializable {
    */
   def nextInt(n: Int): Int = self.nextInt(n)
 
+  /** Returns a pseudorandom, uniformly distributed int value between min
+   *  (inclusive) and the specified value max (exclusive), drawn from this
+   *  random number generator's sequence.
+   */
+  def between(minInclusive: Int, maxExclusive: Int): Int = {
+    require(minInclusive < maxExclusive, "Invalid bounds")
+
+    return nextInt(maxExclusive - minInclusive) + minInclusive
+  }
+
   /** Returns the next pseudorandom, uniformly distributed long value
    *  from this random number generator's sequence.
    */
   def nextLong(): Long = self.nextLong()
 
+   /** Returns a pseudorandom, uniformly distributed long value between min
+   *  (inclusive) and the specified value max (exclusive), drawn from this
+   *  random number generator's sequence.
+   */
+  def between(minInclusive: Long, maxExclusive: Long): Long = ThreadLocalRandom.current.nextLong(minInclusive, maxExclusive)
```

Comment:
As discussed, this doesn't use the seed of this random number generator.  Modulus is a poor solution because for large ranges you can be off by a factor of 2 in probability of getting a value (!).  There are methods that either reroll if you end up off the end, or which use two Longs and do long division by hand to get acceptably small error (2^-64) in probability.